docs/mdbook/book.toml

```diff
@@ -8,3 +8,5 @@ title = "nixvim docs"
 [output.html.fold]
 enable = true
 level = 0
+
+[preprocessor.alerts]
```

Comment:
Thought that was what this was for

Reply:
Yes, looks correct. But in my testing it is rendering a blockquote not an alert

Reply:
I thought it might be a cache issue with the CSS, but hard-reloading with devtools open doesn't help.

Even copying the example verbatim from upstream.

The DOM looks like:
![image](https://github.com/user-attachments/assets/774e4293-43df-4350-ab07-ec91a44f2732)

Reply:
Ah, I think I figured it out. It is the same reason why we can't use headings in option descriptions; mdbook isn't rendering the markdown for the options, the nixpkgs tooling is.

The preprocessor is only running the first part of the page, up until the first option. This first part is effectively a "user-guide" mdbook page, while the rest of the page with options is just concatenated in by us.

Moving the example alerts into a plugin description or a user-guide file will render as:
![image](https://github.com/user-attachments/assets/383d63b0-c480-40b7-942d-bac7df3ac48d)

Reply:
Hm, actually. I'm not quite sure why option descriptions are handled differently.

@traxys did you write these bits?
https://github.com/nix-community/nixvim/blob/e864f2c9e0cda66b0f419b2c3e36374c177d3712/docs/mdbook/default.nix#L230-L248

https://github.com/nix-community/nixvim/blob/e864f2c9e0cda66b0f419b2c3e36374c177d3712/docs/mdbook/default.nix#L265-L273

At first glance, we're just concating raw markdown, and _then_ building. So mdbook shouldn't know what is an option and what is normal md... :confused:

Reply:
Looking at the implementation, this bit is where we write a plugin's markdown file:
https://github.com/nix-community/nixvim/blob/e864f2c9e0cda66b0f419b2c3e36374c177d3712/docs%2Fmdbook%2Fdefault.nix#L199-L202

`pkgs.writeText "module-doc" opts.index.moduleDoc` is working as intended, allowing us to use all markdown features.

`mkMDDoc opts.index.options` is not.

Reply:
`mkMDDoc` is using nixpkgs tooling to get the (commonmark) markdown from the options-json: https://github.com/nix-community/nixvim/blob/e864f2c9e0cda66b0f419b2c3e36374c177d3712/docs%2Fmdbook%2Fdefault.nix#L13-L18

So it's likely that some preprocessing is happening in the nixpkgs function.

Reply:
The processing _may_ involve adding escapes to the \[ and \]

Reply:
(I did not write those, I modified them a bit a while ago)